pkg/convert/convert.go

```diff
@@ -102,13 +103,18 @@ func EncodePropertyMap(enc Encoder, pmap resource.PropertyMap) (tftypes.Value, e
 	return enc.fromPropertyValue(propertyvalue.RemoveSecrets(resource.NewObjectProperty(pmap)))
 }
 
-func DecodePropertyMap(dec Decoder, v tftypes.Value) (resource.PropertyMap, error) {
+func DecodePropertyMap(ctx context.Context, dec Decoder, v tftypes.Value) (resource.PropertyMap, error) {
 	pv, err := dec.toPropertyValue(v)
 	if err != nil {
 		return nil, err
 	}
 	if !pv.IsObject() {
-		return nil, fmt.Errorf("Expected an Object, got: %v", pv.String())
+		details := fmt.Sprintf(`DecodePropertyMap expected the decoder to return an Object
```

Comment:
Emitting a little more information in the debug log. Not returning it as an err in case it is printed on stdout since the values may be sensitive.